src/views/Dashboard.tsx

```diff
@@ -1,21 +1,38 @@
 import * as React from 'react';
-import { useSelector } from 'react-redux';
+import { Redirect } from 'react-router-dom';
 
-import { LogOut, getUserData } from 'modules/user';
-import { SplashSettings, getUserSettings } from 'modules/settings';
-import { ApplicationState } from 'modules/redux-store';
+import { SplashSettings, useSettingsServices } from 'modules/settings';
+import { LogOut, useUserServices } from 'modules/user';
+import {
+  IncrementScore,
+  useScoreListener,
+  useScoreServices,
+} from 'modules/score';
 
 const Dashboard: React.FC = () => {
-  const { userData } = useSelector((state: ApplicationState) =>
-    getUserData()(state),
-  );
-  const { userSettings } = useSelector((state: ApplicationState) =>
-    getUserSettings()(state),
-  );
+  const [{ userData }] = useUserServices();
+  const [{ userSettings }] = useSettingsServices();
+  const [{ userScore }, { getScoreHistory }] = useScoreServices();
+
+  useScoreListener(userData);
+
+  if (!userData) return <Redirect to="/login" />;
 
   return (
     <>
+      <strong>Update score realtime: </strong>
+      {userScore && userScore.score}
+      <IncrementScore user={userData} />
       <SplashSettings userSettings={userSettings} userData={userData} />
+      <strong>Update user history on demand:</strong>
+      <button onClick={getScoreHistory}>Update</button>
+      {userScore?.history &&
+        userScore?.history.length > 0 &&
+        userScore?.history.map(({ score, title }) => (
```

Comment:
userScore?.history?.map

should work as a one-liner. If you are using the map, in the render there is actually no need to check for the length, as it will not fire for empty arrays.

Reply:
Will be removed in the future, just a placeholder for testing